### PR TITLE
Add soft delete (discard) for User model

### LINKED ISSUE #385: Prevent user from deleting account if unpaid invoices issued
At the moment user can destroy their account without any obligations. Destroying user should be prevented if they have won an auction and (unpaid) invoice issued.

### PR DESCRIPTION
**Pull request by yulgolem**
Closes #385 by adding soft-delete (discard) via [discard gem](https://github.com/jhawthorn/discard) on User model.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -35,6 +37,18 @@ class User < ApplicationRecord
...
+    super && !discarded?
+  end
+
+  def can_be_discarded?
+    !cannot_be_discarded?
+  end
+
+  def cannot_be_discarded?
+    invoices.issued.present?
+  end
```
Comment (karlerikounapuu) [inline app/models/user.rb]: What's the purpose of that? Why not combine into single method?

Comment (yulgolem) [reply app/models/user.rb]: Just semantic sugar. I'll change it to single, 'cause the main condition is 'if it can be discarded'

Comment (artur-beljajev) [reply app/models/user.rb]: Perhaps `discardable?` is more eloquent?

Comment (artur-beljajev) [reply app/models/user.rb]: Is there really a need to use gem for such simple feature?

Comment (yulgolem) [reply app/models/user.rb]: discardable? looks good for me.
As for gem - just trying not to reinvent a bike every time.

**Comment by vohmar** [comment]
Artur just pointed out that we already have an audit schema set up for auctions meant to be used for archiving  deleted data. So the question is maybe its not necessary to introduce new  soft delete logic here as similar audit/arhivation model is used in our other apps as well - a bit differently sure (only auction has it set up with separate db schema)

**Comment by yulgolem** [comment]
Well, since User audit via triggers and `audit` namespace is already implemented, the question is - do this PR is needed at all and maybe we should close the issue since deleted users data os already stored?

**Comment by vohmar** [comment]
as user deletion is already handled by audit log solution with records remaining in separate audit db schema I am closing this PR and respective issue ticket. Sorry for the confusion and wasted time and effort.

**Comment by vohmar** [comment]
the soft delete gem was only part of the issue - main point that is to prevent users with unpaid invoices to be deleted should still be solved.

**Comment by yulgolem** [comment]
This PR and soft-delete functionality are not needed.

**Pull request closed without merging**